arch/arm/src/common/Toolchain.defs

```diff
@@ -316,7 +316,7 @@ endif
 
 EXTRA_LIBS += $(COMPILER_RT_LIB)
 
-ifneq ($(CONFIG_LIBM),y)
+ifeq ($(CONFIG_LIBM),y)
   EXTRA_LIBS += $(wildcard $(shell $(CC) $(ARCHCPUFLAGS) --print-file-name=libm.a))
```

Comment:
@centurysys you may misunderstand CONFIG_LIBM. CONFIG_LIBM==true mean use libm provided by nuttx and then skip add libm to EXTRA_LIBS.
To avoid the confusion, it's better to change CONFIG_LIBM from bool to choice(e.g. from CONFIG_LIBM_NONE, CONFIG_LIBM_BUILTIN and CONFIG_LIBM_TOOLCHAIN).

Reply:
It is indeed confusing, I was having trouble with quickjs builds failing.

Reply:
Maybe you can create a new patch to fix the confusion in Kconfig

Reply:
There are so many references to "CONFIG_LIBM", it might be better to rename it as "CONFIG_LIBM_BUILTIN".